src/main/java/com/microsoft/graph/content/MSBatchRequestContent.java

```diff
@@ -37,16 +39,32 @@
      * Maximum number of requests that can be sent in a batch
      */
     public static final int MAX_NUMBER_OF_REQUESTS = 20;
+    private final ILogger logger;
 
     /**
      * Creates Batch request content using list provided
      *
      * @param batchRequestStepsArray List of batch steps for batching
      */
     public MSBatchRequestContent(@Nonnull final MSBatchRequestStep... batchRequestStepsArray) {
+        this(new DefaultLogger(), batchRequestStepsArray);
+    }
+
+    /**
+     * Creates Batch request content using list provided
+     *
+     * @param batchRequestStepsArray List of batch steps for batching
+     * @param logger logger to use for telemetry
+     */
+    public MSBatchRequestContent(@Nonnull final ILogger logger, @Nonnull final MSBatchRequestStep... batchRequestStepsArray) {
         if (batchRequestStepsArray.length > MAX_NUMBER_OF_REQUESTS)
             throw new IllegalArgumentException("Number of batch request steps cannot exceed " + MAX_NUMBER_OF_REQUESTS);
 
+        if(logger == null)
```

Comment:
You might want to consider using `Objects.requireNonNull()` here. With a static import, it'd look pretty concise. But if there's a special for this way of handling the null value, then I suggest adding a space after `if`.

Reply:
thanks, I didn't know that static method existed. in dotnet we have the null coalescing operator that allows us to do those sort of things using a very succinct syntax
```CSharp
myProp = myParam ?? throw new ArgumentNullException(nameof(myParam));
```

Reply:
Java's still better.: )

Reply:
I guess that's a preference thing, the null coalesce let's me choose what kind of exception I want to throw, it also let's me set a default value. Non of that seems possible with the requireNonNull static method.

Reply:
Yeah, the null coalesce is much more concise. The Java language's own syntax (at least to my humble knowledge) doesn't offer much more, but some of are libraries which are de facto standards and "language extensions" and offer that.